Content.Shared/Actions/SharedActionsSystem.cs

```diff
@@ -416,21 +316,17 @@ private void OnActionRequest(RequestPerformActionEvent ev, EntitySessionEventArg
         if (!action.Enabled)
             return;
 
+        var curTime = GameTiming.CurTime;
+        if (IsCooldownActive(action, curTime))
+            return;
+
         // check for action use prevention
         // TODO: make code below use this event with a dedicated component
         var attemptEv = new ActionAttemptEvent(user);
         RaiseLocalEvent(actionEnt, ref attemptEv);
         if (attemptEv.Cancelled)
             return;
 
-        var curTime = GameTiming.CurTime;
-        if (IsCooldownActive(action, curTime))
-            return;
-
-        // TODO: Replace with individual charge recovery when we have the visuals to aid it
-        if (action is { Charges: < 1, RenewCharges: true })
-            ResetCharges(actionEnt, true, true);
-
```

Comment:
This makes an unintended huge nerf.
If you have something with 2 charges and a use delay, the action fires off, so does the use delay, and then you're unable to use an action _until_ the use delay is up. 

The behavior should be:
If an action still has charges remaining, use delay should not trigger until the charges are used up.

Reply:
Charges didn't work before and I'm not super interested in design changes to it in the same PR, I'm mainly worried about the big performance issues charges are causing atm. This is effectively a selective revert without having to completely remove everything.

Reply:
We need some sort of information to the player that a charge is recharging. Even if it's just a timer on the examine text.

And logic should be:
Use delay is never longer/equal to the recharge timer.